s2aff/timo/interface.py

```diff
@@ -0,0 +1,146 @@
+    child: List[str] = Field(description="Child affiliation strings from NER step")
+    address: List[str] = Field(description="Address affiliation strings from NER step")
+
+
+class PredictorConfig(BaseSettings):
+    """
+    Configuration required by the model to do its work.
+    Uninitialized fields will be set via Environment variables.
+
+    The fields below are examples only; please replace them with ones
+    appropriate for your model. These serve as a record of the ENV
+    vars the consuming application needs to set.
+    """
+
+
+class Predictor:
+    """
+    Interface on to your underlying model.
+
+    This class is instantiated at application startup as a singleton.
+    You should initialize your model inside of it, and implement
+    prediction methods.
+
+    If you specified an artifacts.tar.gz for your model, it will
+    have been extracted to `artifacts_dir`, provided as a constructor
+    arg below.
+    """
+
+    _config: PredictorConfig
+    _artifacts_dir: str
+
+    def __init__(self, config: PredictorConfig, artifacts_dir: str):
+        self._config = config
+        self._artifacts_dir = artifacts_dir
+        self._load_model()
+
+    def _load_model(self) -> None:
+        """
+        Perform whatever start-up operations are required to get your
+        model ready for inference. This operation is performed only once
+        during the application life-cycle.
+        """
+        ner_predictor = NERPredictor(
+            model_path=join(self._artifacts_dir, basename(PATHS["ner_model"])), use_cuda=torch.cuda.is_available()
+        )
+        ror_index = RORIndex(
+            ror_data_path=join(self._artifacts_dir, basename(PATHS["ror_data"])),
+            country_info_path=join(self._artifacts_dir, basename(PATHS["country_info"])),
+            works_counts_path=join(self._artifacts_dir, basename(PATHS["openalex_works_counts"])),
+        )
+        pairwise_model = PairwiseRORLightGBMReranker(
+            ror_index,
+            model_path=join(self._artifacts_dir, basename(PATHS["lightgbm_model"])),
+            kenlm_model_path=join(self._artifacts_dir, basename(PATHS["kenlm_model"])),
+        )
+        self.s2aff = S2AFF(ner_predictor, ror_index, pairwise_model)
+
+    @staticmethod
+    def convert_raw_prediction_to_Prediction(prediction) -> Prediction:
+        if len(prediction["stage2_candidates"]) == 0:
+            prediction_instance = Prediction(ror_id="", score=0, main="", child="", address="")
+        else:
+            prediction_instance = Prediction(
+                ror_id=prediction["stage2_candidates"][0],
+                score=prediction["stage2_scores"][0],
+                main=prediction["main_from_ner"],
+                child=prediction["child_from_ner"],
+                address=prediction["address_from_ner"],
+            )
+        return prediction_instance
+
+    def predict_one(self, instance: Instance) -> Prediction:
```

Comment:
If your'e not calling through to `predict_one` from `predict_batch`, you can drop it altogether. The server code will always call `predict_batch`.

Reply:
Maybe add this fact to the doc string for `predict_one`?